Sanitize node response POJO (#754)
gentics/mesh: CHANGELOG.adoc

* The support for the embedded Elasticsearch will be dropped in the future. It is highly recommended to link:{{< relref 
"elasticsearch.asciidoc" >}}#_dedicated_elasticsearch[setup Elasticsearch as a dedicated service].

[[v0.35.0]]
== 0.35.0 (TBD)

> CAUTION: Java Rest Client: The default field values for `NodeResponse` and `AbstractGenericRestResponse` has been updated. 

Reviewer: ```suggestion
CAUTION: Java Rest Client: The default field values for `NodeResponse` and `AbstractGenericRestResponse` have been updated. 
```

The following values have been changed:

* `AbstractGenericRestResponse#permissions`
* `NodeResponse#tags`
* `NodeResponse#childrenInfo`
* `NodeResponse#container`
* `NodeResponse#fields`
* `NodeResponse#breadcrumb`

The `NodeResponse#isContainer` method has been deprecated and was replaced by `NodeResponse#getContainer`

> Applications which use affected models will most likly not needed to updated since the returned response will always set the affected fields.

Reviewer: ```suggestion
Applications which use affected models will most likely not need to updated since the returned response will always set the affected fields.
```

> icon:plus[] REST: The `?fields` query parameter which can be used to filter response properties will now work more granular on node responses.

Reviewer: ```suggestion
icon:plus[] REST: The `?fields` query parameter which can be used to filter response properties will now work more granularly on node responses.
```
